### DISCUSSION TITLE
XIAO ESP32C3 support

### DISCUSSION THREAD

**Discussion by Frida854** [Moddable-OpenSource/moddable#1101] [Q&A] [answered]
I've gotten hold of a handful of these, but I'm having trouble getting them to work. They compile fine and download, but then it stops.
Is it possible to make them work under Moddable?

I tried the Hello World example with
'mcconfig -d -m -p esp32/esp32_c3'
on a mac computer.

[https://wiki.seeedstudio.com/XIAO_ESP32C3_Getting_Started]

<img width="578" alt="pic01" src="https://user-images.githubusercontent.com/18483582/236283185-d834090a-8406-46e9-8e99-0d7095ae996c.png">

**Comment by phoddie** [comment]
I just tried a build with `mcconfig -d -m -p esp32/esp32c3` (same as your console window title shows) using the Espressif ESP32-C3-DevKitM-1. That built, installed, and ran normally. The end of my console output is the same as yours, up to the "Done":

```
Writing at 0x00008000... (100 %)
Wrote 3072 bytes (83 compressed) at 0x00008000 in 0.1 seconds (effective 211.9 kbit/s)...
Hash of data verified.

Leaving...
Hard resetting via RTS pin...

Detecting chip type... ESP32-C3
Running ninja in directory /Users/hoddie/Projects/moddable/build/tmp/esp32/esp32c3/debug/helloworld/xsProj-esp32c3/build
Executing "ninja flash"...
Done
```

You appear fall into the console where I have no further output as `serial2xsbug` is running to bridge the serial port to xsbug.

But... I'm not familiar with the XIAO ESP32C3 board. Perhaps it requires a different configuration?

**Reply by mkellner** [reply, marked as answer]
The `esp32/esp32c3` target uses the on-board serial to usb converter.

In the next code-drop there will be support for `USE_USB` on esp32c3 devices.
There will be some new targets including `xiao_esp32c3` and `qtpyc3`.

**Reply by LokiMetaSmith** [reply]
I'm unreasonably excited about this, since Twig supports this.
